spinalcordtoolbox/download.py

```diff
@@ -40,7 +40,9 @@
     },
     "sct_testing_data": {
         "mirrors": [
-            "https://github.com/spinalcordtoolbox/sct_testing_data/releases/download/r20230207/sct_testing_data-r20230207.zip",
+            # TODO: Replace with stable release generated by distriploy once PR is merged:
+            #       https://github.com/spinalcordtoolbox/sct_testing_data/pull/12
+            "https://github.com/spinalcordtoolbox/sct_testing_data/archive/jn/update-for-sct-4302-4345.zip",
```

Comment:
I've updated the `sct_testing_data` link with the changes from https://github.com/spinalcordtoolbox/sct_testing_data/pull/12 to improve the tests.

TODO for myself: Once we're happy with this PR, I can merge the changes upstream, then create a new stable release of `sct_testing_data`.

This might be a bit tricky, though, given that #4302 also requires changes to `sct_testing_data`. Perhaps what I'll do is merge #4302 with the "temp" dataset, then we can merge this PR with the "real" dataset (since I imagine this PR will take longer to finalize).

Reply:
Dataset updated in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/418e504e9ef0056ff763d26c316044d993888cbb!